2-0-data-structures-and-algorithms/src/test/java/com/bobocode/array_list/ArrayListTest.java

```diff
@@ -251,20 +315,34 @@ public void setFirstElementOnEmptyTree() {
     @Test
     @Order(24)
     public void removeElementByIndex() {
-        arrayList = ArrayList.of(15, 69, 58, 78, 100);
+        try {
+            getFieldByName("elementData").set(arrayList, new Object[]{15, 69, 58, 78, 100});
+            getFieldByName("size").set(arrayList, 5);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
 
         int removedElement = arrayList.remove(2);
 
-        assertThat(arrayList.get(2)).isEqualTo(78);
-        assertThat(arrayList.get(1)).isEqualTo(69);
-        assertThat(arrayList.size()).isEqualTo(4);
+        Object[] elements = (Object[]) getObjectByName("elementData");
+        int size = (int) getObjectByName("size");
+
+        assertThat(elements[2]).isEqualTo(78);
+        assertThat(elements[1]).isEqualTo(69);
+        assertThat(size).isEqualTo(4);
         assertThat(removedElement).isEqualTo(58);
     }
 
     @Test
     @Order(25)
     public void removeElementByIndexEqualToSize() {
```

Comment:
@erneman1 let's remant test method to `removeElementByIndexThrowsExceptionWhenIndexEqualsSize`